### ISSUE TITLE
Kerning went wrong in Variable Font?

### ISSUE THREAD

**Issue by Mark2Mark** [hyper-type/hahmlet#8]
I just spotted an obviously missing kerning in the variable font, where "We" are too far appart. I am sure to have kerned that in the original file, and have not time to check right now. So I'll keep this here as a reminder ...

**Comment by ErwinDenissen** [comment]
All fonts lack pair adjustment positioning.

**Comment by Mark2Mark** [comment]
@aaronbell @davelab6 where is the kerning gone to? There is a massive amount of kerning in the source file.

**Comment by ErwinDenissen** [comment]
The source file lacks class definitions.

**Comment by Mark2Mark** [comment]
I don’t think so:
<img width="816" alt="Screenshot 2021-08-07 at 10 01 38" src="https://user-images.githubusercontent.com/5824595/128593350-6ca2aedd-930f-44a4-bc9c-b105574eef63.png">

**Comment by ErwinDenissen** [comment]
I am talking about @MMK_L_W and @MMK_R_o, but maybe I just don't know how to interpret the source file.

**Comment by aaronbell** [comment]
Well, that's weird. The kerning is being brought over to the UFOs, but for some reason not being added when the font files are built. I've taken a look through my build script but can't see anything that would cause kerning to be dropped. Perhaps the version of fontmake I was using is old? Not sure. 

Out of interest, I converted the repro over to the newer [Unified Font Repository](https://github.com/unified-font-repository/Unified-Font-Repository) structure and it appears that the font builds now with kerning. I've got a few more things to fix up on it, and then will submit a PR on it.

**Comment by Mark2Mark** [comment]
Thanks sooo much, @aaronbell 

Keep us posted if/how we can help 🙏

**Comment by aaronbell** [comment]
Hi @Mark2Mark Sorry for the delay—last week went a bit sideways...

I've pushed a new version of the repro and fonts to the server. I compared it with the current version on google fonts and think things are looking good, but if you spot anything that you'd like me to change, let me know! Otherwise, I'll PR the updated fonts to GF.

**Comment by aaronbell** [comment]
(oh, one thing I should note. With the new system, there is an automated Github Action that will build the fonts on the server on PR. You can download the updated fonts as artifacts from the Actions tab. [For example](https://github.com/hyper-type/hahmlet/actions/runs/1136501580))

**Comment by Mark2Mark** [comment]
Hi @aaronbell 


# First
Thanks so much for taking care of the build and kerning.

I had a look at the files and random sampled several instances of the OTFs and TTFs, as well as the variable font, and it appears as that the kerning is back and running 🎉

Do you have some Google recommendations for what else we could double check (manually)? We had suspected that fontBakery takes care of the important things, we could not have expected that it might not detect missing kerning.
So for now one thing I also checked once more is the localisation (e.g. `lang="ca"` for Catalan `l·l`)
I am aware of the Google Fonts checklist, we where working trhrough it during design and engineering phases, but maybe you got some proofing/checking workflows on top of that?


# Second

I am not sure if I understand your linked example, how it came into being. Is there a documentation about how to set up such an action for Google Fonts? For now I just take it as a note and am happy to see it working.

# Lastly
Please let us know when the new build are live in the Google Fonts Universe 😃

**Comment by aaronbell** [comment]
Thanks Mark! 

1) The general approach that Google Fonts used is, unfortunately a generally manual one where I set up the build script, you reviewed it, and two of my colleagues reviewed it (including a text string test) before it was pushed live. Fontbakery did actually note that the font didn't have kerning, but I think it got overlooked as the Fontbakery's CJK support is a little so-so and I probably got confused thinking that it was complaining about the Hangul, which generally doesn't need kerning. 

The general testing tool that googlefonts uses is gftools qa, but that's used primarily for comparing two fonts, rather than evaluating a new font. It does have some tools built in for generating the text strings, but I think that's reliant on browserstack. 

Generally, I think the expectation is that you'd run the final version through all of your testing sheets and make sure everything is working alright. On the plus side, I think the font renders pretty well, even without kerning! :)

2) Github Actions are actually more general than Google Fonts. I uploaded the workflow build file here: 
https://github.com/hyper-type/hahmlet/blob/master/.github/workflows/build.yaml
There's more info available here: https://github.com/googlefonts/Unified-Font-Repository

3) Will do!
